### ISSUE TITLE
Rule: Exists

### ISSUE THREAD

**Issue by tobiastimm** [loeffel-io/ls-lint#30]
It would be cool if this would have a check to ensure that a directory/file exists!

**Comment by loeffel-io** [comment]
Yea, also thought about this but its not that easy to implement 
Will put this on the wishlist

**Comment by loeffel-io** [comment]
How you would solve this in your config?

**Comment by tobiastimm** [comment]
Would be cool to have something like
```
ls:
    packages/src:
        .dir: 
             name: /REGEX_PATTERN/
             exists: true
```
Pretty similar to how [goss](https://github.com/aelsabbahy/goss) handles that
```
user:
  sshd:
    exists: true
    uid: 74
    gid: 74
    groups:
    - sshd
    home: /var/empty/sshd
    shell: /sbin/nologin
```

**Comment by loeffel-io** [comment]
like it but conflicts with my current idea of handling files via `exists` rule: 

maybe we could add .dir to this list like: 

`.dir: exists:.js,.ts,.dir`

Maybe you have an better idea

**Comment by loeffel-io** [comment]
Yeah, thats what i also thought about 👍

**Comment by Asuza** [comment]
I was just wondering if this feature is still on the table? This is exactly what I need to ensure some yarn files don't accidentally get removed.

**Comment by loeffel-io** [comment]
still on the table - prs are welcome!

**Comment by arei** [comment]
+1

**Comment by ccoVeille** [comment]
For anyone interested. You can now use `exists:1` as a rule. Please check the documentation

**Comment by loeffel-io** [comment]
@ccoVeille `v2.3.0` is not released yet - there is one single piece that is missing

**Comment by ccoVeille** [comment]
Author opened the issue 4 years ago. I assumed he could wait a few more days/week.

But you are right, it's not yet available or documented

**Comment by loeffel-io** [comment]
As far i can tell this is one of the most challenging tasks for me - its beautiful how wildcard extensions, selected files and exists working together magically

**Comment by loeffel-io** [comment]
This feature is now available in the `v2.3.0-beta.1` release. 
Please test it and let me know if there are any issues ❤️ 

The upcoming v2.3.0 blog post: https://ls-lint.org/blog/announcements/v2.3.0.html
The docs: https://ls-lint.org/2.3/getting-started/introduction.html
The release notes: https://github.com/loeffel-io/ls-lint/releases/tag/v2.3.0-beta.1
